storage: add a "low disk space" event

RaduBerinde opened the pull request:
Post an event (visible in the Events page) when a store is running low
on disk space. We post an event the first time when the available
bytes go below 10%, then when they go below 5%, then 3%, 2%, 1%. We
repost the event every half hour until the condition is resolved.

Fixes #129319
Release note (ui change): an event is posted when a store is getting
close to full capacity.

---

Example in the UI:

<img width="921" alt="image" src="https://github.com/user-attachments/assets/1d82a915-0e2e-425f-acb5-36b7d882c096" />

It also shows up in logs:
```
I250116 04:35:29.718164 83 3@pebble/event.go:1017 ⋮ [n?,s?,pebble] 23  available disk space under 10% (36GB of 368GB)
I250116 04:40:21.377270 10391 3@pebble/event.go:1017 ⋮ [n2,s2,pebble] 145  available disk space under 5% (18GB of 368GB)
```

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139199)
<!-- Reviewable:end -->

RaduBerinde committed:
storage: add a "low disk space" event

Post an event (visible in the Events page) when a store is running low
on disk space. We post an event the first time when the available
bytes go below 10%, then when they go below 5%, then 3%, 2%, 1%. We
repost the event every half hour until the condition is resolved.

Fixes #129319
Release note (ui change): an event is posted when a store is getting
close to full capacity.

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 8 of 10 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139199#-:-OGqG8Fe23qsMr-0jRh8:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @aa-joshi, @aadityasondhi, @arjunmahishi, @dhartunian, and @xinhaoz)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

bors r+

RaduBerinde commented:
bors r+
